core/src/test/scala/kafka/common/InterBrokerSendThreadTest.scala

```diff
@@ -51,6 +52,26 @@ class InterBrokerSendThreadTest {
         Some(queue.dequeue())
       }
     }
+    override def pollOnce(maxTimeoutMs: Long): Unit = {
+      try super.pollOnce(maxTimeoutMs)
+      catch {
+        case e: Throwable => exceptionCallback(e)
+      }
+    }
+
+  }
+
+  @Test
+  def shutdownThreadShouldNotCauseException(): Unit = {
```

Comment:
I don't think we should have a test inside a non test class. Gradle won't run it for once and it's quite confusing.

Reply:
@ijuma Pardon me. I failed to catch your point. ```InterBrokerSendThreadTest.scala``` is a true test case and it seems to me the test make sure InterBrokerSendThreadTest does not throw fatal exception when it is shutdown correctly.

Reply:
My bad, I thought this was `TestInterBrokerSendThread`, but that's an inner class.